### PR TITLE
registry: allow to customize cache size for DELETE

### PR DESCRIPTION
**Pull request by fyrchik**
Signed-off-by: Evgenii Stratonikov <stratonikov@runbox.com>

### CODE REVIEW

**Review by alexvanin — APPROVED** [review]

File: internal/registry/obj_selector.go [hunk 1]
```diff
@@ -15,24 +15,29 @@ type ObjFilter struct {
...
-	filter  *ObjFilter
+	ctx       context.Context
+	objChan   chan *ObjectInfo
+	boltDB    *bbolt.DB
+	filter    *ObjFilter
+	cacheSize int
 }
 
-// objectSelectCache is a maximum number of the selected objects to be
+// objectSelectCache is a default number of the selected objects to be
```
Comment (carpawell) [inline internal/registry/obj_selector.go:25]: "maximum" was not added by accident, it is an expected situation when there are not enough objects to fill the cache

File: scenarios/grpc.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ if (registry_enabled && delete_age) {
     obj_to_delete_selector = registry.getSelector(
         __ENV.REGISTRY_FILE,
         "obj_to_delete",
+        __ENV.SELECTION_SIZE ? parseInt(__ENV.SELECTION_SIZE) : 0,
```
Comment (carpawell) [inline scenarios/grpc.js:33]: could you recheck that it is the only JS `getSelector` call?

**Review by carpawell — APPROVED** [review]

File: scenarios/grpc.js (hunk 2 above)
Comment (fyrchik) [reply scenarios/grpc.js:33]: Fixed

File: internal/registry/obj_selector.go (hunk 1 above)
Comment (fyrchik) [reply internal/registry/obj_selector.go]: Rephrased a bit

**Review by carpawell — APPROVED** [review]

**Review by alexvanin — DISMISSED** [review]

**Comment by alexvanin** [comment]
@vdomnich-yadro maybe you would like to see this change too.

**Comment by fyrchik** [comment]
Also added logging!
```
INFO[0001] {"cid":"9sVMdHhH7GGfaMCn6TGAkn4Hak68sELMwPvZVMZ6RUzX","oid":"6MpoHt7FnCc4d3CsSn88utiCeEfQfaFbbNTqwXRtoKxW","error":"status: code = 3072 message = container not found"}  source=console
```

**Review by alexvanin — APPROVED** [review]

File: scenarios/verify.js [hunk 3]
```diff
@@ -70,7 +71,11 @@ export const options = {
 export function setup() {
     // Populate counters with initial values
     for (const [status, counter] of Object.entries(obj_counters)) {
-        const obj_selector = registry.getSelector(__ENV.REGISTRY_FILE, status, { status });
+        const obj_selector = registry.getSelector(
+            __ENV.REGISTRY_FILE,
+            status,
+            __ENV.SELECTION_SIZE ? parseInt(__ENV.SELECTION_SIZE) : 0,
```
[the remark this answers is not in the document]
Comment (fyrchik) [reply scenarios/verify.js:77]: Fixed

**Pull request merged**